ansible_runner/utils.py

```diff
@@ -83,9 +84,10 @@ def check_isolation_executable_installed(isolation_executable):
         return False
 
 
-def stream_dir(directory):
-    buf = BytesIO()
-    with zipfile.ZipFile(buf, 'w', compression=zipfile.ZIP_DEFLATED, allowZip64=True) as archive:
+def stream_dir(directory, buf):
+    fd, zip_file = tempfile.mkstemp(suffix='.zip')
```

Comment:
I'd suggest `tempfile.TemporaryFile` instead so that the zipfile is always deleted, even if something goes wrong before the end of this function.

Reply:
(or maybe `NamedTemporaryFile`)